CMakeLists.txt

```diff
@@ -722,32 +711,39 @@ if(WITH_OPTIM)
         endif()
         if(WITH_AVX2 AND HAVE_AVX2_INTRIN)
             add_definitions(-DX86_AVX2 -DX86_AVX2_ADLER32)
-            list(APPEND ZLIB_ARCH_SRCS ${ARCHDIR}/slide_avx.c ${ARCHDIR}/adler32_avx.c)
+            list(APPEND ZLIB_ARCH_SRCS ${ARCHDIR}/slide_avx.c)
+            set(AVX2_SRCS ${ARCHDIR}/slide_avx.c)
             add_feature_info(AVX2_SLIDEHASH 1 "Support AVX2 optimized slide_hash, using \"${AVX2FLAG}\"")
-            list(APPEND ZLIB_ARCH_SRCS ${ARCHDIR}/compare258_avx.c)
+            list(APPEND AVX2_SRCS ${ARCHDIR}/compare258_avx.c)
             add_feature_info(AVX2_COMPARE258 1 "Support AVX2 optimized compare258, using \"${AVX2FLAG}\"")
+            list(APPEND AVX2_SRCS ${ARCHDIR}/adler32_avx.c)
             add_feature_info(AVX2_ADLER32 1 "Support AVX2-accelerated adler32, using \"${AVX2FLAG}\"")
-            add_intrinsics_option("${AVX2FLAG}")
+            list(APPEND ZLIB_ARCH_SRCS ${AVX2_SRCS})
+            add_intrinsics_option("${AVX2FLAG}" ${AVX2_SRCS})
         endif()
         if(WITH_SSE4 AND (HAVE_SSE42CRC_INLINE_ASM OR HAVE_SSE42CRC_INTRIN))
             add_definitions(-DX86_SSE42_CRC_HASH)
-            list(APPEND ZLIB_ARCH_SRCS ${ARCHDIR}/insert_string_sse.c)
+            set(SSE42_SRCS ${ARCHDIR}/insert_string_sse.c)
             add_feature_info(SSE42_CRC 1 "Support SSE4.2 optimized CRC hash generation, using \"${SSE4FLAG}\"")
-            add_intrinsics_option("${SSE4FLAG}")
+            list(APPEND ZLIB_ARCH_SRCS ${SSE42_SRCS})
+            add_intrinsics_option("${SSE4FLAG}" ${SSE42_SRCS})
             if(HAVE_SSE42CRC_INTRIN)
                 add_definitions(-DX86_SSE42_CRC_INTRIN)
             endif()
-        endif()
+            endif()
         if(HAVE_SSE42CMPSTR_INTRIN)
             add_definitions(-DX86_SSE42_CMP_STR)
-            list(APPEND ZLIB_ARCH_SRCS ${ARCHDIR}/compare258_sse.c)
+            set(SSE42_SRCS ${ARCHDIR}/compare258_sse.c)
             add_feature_info(SSE42_COMPARE258 1 "Support SSE4.2 optimized compare258, using \"${SSE4FLAG}\"")
+            list(APPEND ZLIB_ARCH_SRCS ${SSE42_SRCS})
+            add_intrinsics_option("${SSE4FLAG}" ${SSE42_SRCS})
         endif()
         if(WITH_SSE2 AND HAVE_SSE2_INTRIN)
-            add_definitions(-DX86_SSE2)
-            list(APPEND ZLIB_ARCH_SRCS ${ARCHDIR}/slide_sse.c)
+            add_definitions(-DX86_SSE2 -DX86_SSE2_MEMCHUNK)
+            set(SSE2_SRCS ${ARCHDIR}/memchunk_sse.c ${ARCHDIR}/slide_sse.c)
+            list(APPEND ZLIB_ARCH_SRCS ${SSE2_SRCS})
             if(NOT ${ARCH} MATCHES "x86_64")
-                add_intrinsics_option("${SSE2FLAG}")
+                add_intrinsics_option("${SSE2FLAG}" ${SSE2_SRCS})
```

Comment:
configure seems to build memchunk_sse.c with different options than cmake.

configure uses -msse4
cmake doesn't.

What's up with that?  7e8593bd may have a thinko in arch/x86/Makefile.in...?

Found with https://github.com/dankegel/zlib-ng/tree/cmake-intrinsics-identical like so:
```
  sh test/pkgcheck.sh
```

And for the gory details:
```
  sh test/pkgcheck.sh > log 2>&1
  grep 'memchunk_sse.c$' < log  | grep -v fPIC > x; head -n1 < x | tr ' ' '\012' | grep . | sort > x1; tail -n1 < x | tr ' ' '\012' | grep . | sort > x2; diff -u x1 x2
```

Reply:
Good find! I have fixed _arch/x86/Makefile.in_ to use SSE2 flag.